Sources/Datasets/Batch.swift

```diff
@@ -1,10 +1,12 @@
-protocol S5TFBatch {}
+import TensorFlow
+
+public protocol S5TFBatch {}
 
 public struct S5TFUnlabeledBatch: S5TFBatch {
     var data: Float
 }
 
 public struct S5TFLabeledBatch: S5TFBatch {
-    var data: Int
-    var labels: Int
+    var data: Tensor<Float>
+    var labels: Tensor<Int32>
```

Comment:
The types of `data` and `labels` are quite specific to the Iris dataset... for example MNIST has `data` as `Tensor<UInt8>` and `labels` as `Tensor<Int64`.

Reply:
I think all values need to be downsampled to 0...1 when training, at least in computer vision.

I agree that this might not extend very well to other datasets. We could make something like this:

```swift
public struct S5TFLabeledBatch<T: TensorFlowScalar>: S5TFBatch {
    var data: Tensor<T>
    var labels: Tensor<Int32>
}
```

Working demo: https://colab.research.google.com/drive/16U-6MeGmOZ9O4ik0jgtAkAk-dINVzDrq#scrollTo=sRTUKjuTUXo2

It would be even better to add a constraint like that to `S5TFBatch`, but that approach also has drawbacks. For example, it would be hard to specify what kind of batch a `DataLoader` would return and we probably don't want to specify a generic paramater on `DataLoader` itself, but that could be an interesting idea. `DataLoader<Float>(...)`

Labels always need to be `Int32` because that's what S4TF expects in their loss functions.

Reply:
I'm thinking whether or not we should include an option to keep the raw data: for example, TFDS returns the raw data and it's up to the user to normalize (for example, in MNIST taking the `Tensor<UInt8>` and /255 to convert it to `Float`, similarly for the labels. (Check the TFDS Python interop, we load the raw data there)

Reply:
Could you please make that Google Colab public (link share)? Thanks

Reply:
True. I have an idea on how to solve this that I will describe in more detail in a second below.

Reply:
This will be solved in a future PR. See comment https://github.com/s5tf-team/datasets/pull/14#issuecomment-578440647.